app/models/katello/package_group.rb

```diff
@@ -29,5 +29,9 @@ def package_names
       group = service_class.new(self.pulp_id)
       group.default_package_names + group.conditional_package_names + group.optional_package_names + group.mandatory_package_names
     end
+
+    def content_view_filters
+      Katello::ContentViewPackageGroupFilterRule.where(uuid: self.pulp_id).map(&:filter)
```

Comment:
I think I'm OK w/ this although it looks like there's an N+1 situation now, which was one of the benefits of using the associations. I think you can solve that by throwing `.eager_load(:filter)` on the query which will create the join

Reply:
That's a good point, I added it here.

FWIW I played around with this and came up with a query that I think works for what we are trying to do:
```SQL
SELECT kcvf.id
FROM katello_content_view_filters AS kcvf
INNER JOIN katello_content_views AS kcv ON kcv.id = kcvf.content_view_id
INNER JOIN katello_content_view_package_group_filter_rules AS kcvpgfr ON kcvpgfr.content_view_filter_id = kcvf.id
INNER JOIN katello_package_groups AS kpg ON kcvpgfr.uuid = kpg.pulp_id
WHERE kpg.id = 2;
```
(replace 2 with the package group id), its pretty similar with to what you suggested, but the trouble is "railsifying" it.

I'm ok with whats there now, but if you think it's worth it we can figure out how to get the stars aligned with the associations :star2:

Reply:
That query looks accurate. We've gone this far without having the associations I was looking to have added, and your approach is more straightforward so I'm happy to stick with it. Will give it a test today